### ISSUE TITLE
FloatingLabel input cuts label on Android

### ISSUE THREAD

**Issue by lucianomlima** [GeekyAnts/NativeBase#2117]
iOS works fine.

```json
    "native-base": "^2.7.0",
    "react": "16.3.1",
    "react-native": "0.55.4",
```

![Android screenshot](https://user-images.githubusercontent.com/1580205/42468106-44ed47de-838a-11e8-9bd5-49928121d4d3.png)

![iOS screenshot](https://user-images.githubusercontent.com/1580205/42468135-5c4a1dda-838a-11e8-9c7d-5608f4933baf.png)

My code:

```javascript
// @flow
import React, { Component } from 'react';
import { commitMutation, graphql } from 'react-relay';
import images from '~/resources/images';
import { connect } from 'react-redux';

import {
  Alert,
  AsyncStorage,
  Image,
  Keyboard,
  KeyboardAvoidingView,
  Platform,
  StyleSheet,
  Text,
  TouchableWithoutFeedback,
  View,
} from 'react-native';

import {
  Container,
  Button,
  Form,
  Item,
  Input,
  Label,
} from 'native-base';

import { NavigationActions } from 'react-navigation';
import constants from '~/constants';

type State = {
  email: string,
  password: string,
  isSubmiting: boolean,
  hasLoginError: boolean,
}

type Props = {
  dispatch: DispatchFn,
  navigation: Object,
}

const spacer = 8;
const logoHeight = spacer * 5;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: 'white',
    justifyContent: 'center',
    alignItems: 'center',
  },
  logoContainer: {
    alignItems: 'center',
  },
  welcomeImage: {
    width: spacer * 18,
    height: logoHeight,
    resizeMode: 'contain',
    justifyContent: 'center',
    alignItems: 'center',
  },
  loginWrapper: {
    justifyContent: 'center',
  },
  btnWrapper: {
    marginTop: spacer * 4,
  },
  formWrapper: {
    width: constants.device.width - (spacer * 4),
  },
  formItem: {
    marginLeft: 0,
  },
  buttonText: {
    color: 'white',
    fontSize: 16,
  },
});

export class LoginScreen extends Component<Props, State> {
  passwordInput: null

  constructor(props: Props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      isSubmiting: false,
      hasLoginError: false,
    };
  }

  render() {
    const { email, password, isSubmiting } = this.state;
    const behavior = Platform.select({ android: undefined, ios: 'padding' });

    return (
      <KeyboardAvoidingView style={styles.container} behavior={behavior}>
        <TouchableWithoutFeedback onPress={this.blurForm}>
          <Container style={styles.loginWrapper}>
            <View style={styles.logoContainer}>
              <Image
                source={images.jusbrasilLogo}
                style={styles.welcomeImage}
              />
            </View>

            <Form style={styles.formWrapper}>
              <Item floatingLabel style={styles.formItem}>
                <Label>E-mail ou usuário</Label>
                <Input
                  autoCapitalize="none"
                  autoCorrect={false}
                  value={email}
                  blurOnSubmit={false}
                  keyboardType="email-address"
                  returnKeyType="next"
                />
              </Item>

              <Item floatingLabel style={styles.formItem}>
                <Label>Senha</Label>
                <Input
                  autoCapitalize="none"
                  secureTextEntry
                  returnKeyType="done"
                  value={password}
                />
              </Item>
              <View style={styles.btnWrapper}>
                <Button
                  full
                  success
                  disabled={isSubmiting || !(email && password)}
                  onPress={this.tryLogin}
                >
                  <Text style={styles.buttonText}>{isSubmiting ? 'Entrando...' : 'Entrar'}</Text>
                </Button>
              </View>
            </Form>
          </Container>
        </TouchableWithoutFeedback>
      </KeyboardAvoidingView>
    );
  }
}

const LoginStoreAware = connect()(LoginScreen);

LoginStoreAware.navigationOptions = () => ({
  header: null,
});

export default LoginStoreAware;
```

**Comment by SupriyaKalghatgi** [comment]
Worked fine for me
Try just `floatingLabel` input without styles

**Comment by lucianomlima** [comment]
@SupriyaKalghatgi I will try and come here with results. Thanks.

**Comment by lucianomlima** [comment]
@SupriyaKalghatgi same problem. I create a clean project with lastest native base and RN 0.55.4
Take a look please: https://github.com/lucianomlima/nativebase-floatinglabel-android-bug

**Comment by SupriyaKalghatgi** [comment]
ok, i will check this

**Comment by SupriyaKalghatgi** [comment]
@lucianomlima I cloned your [repo](https://github.com/lucianomlima/nativebase-floatinglabel-android-bug), worked fine
![image](https://user-images.githubusercontent.com/13464885/42552617-d23e9374-84fa-11e8-8859-897c090d387b.png)

Also tried this use-case, Label text in capitals
![image](https://user-images.githubusercontent.com/13464885/42552675-1d660198-84fb-11e8-93b3-86473e2c5ec1.png)

**Comment by SupriyaKalghatgi** [comment]
Tested on device
Nexus 6, Android 7.1.1

**Comment by lucianomlima** [comment]
Maybe only occurs in Android 8?I Will test in another device with other Android version.

**Comment by SupriyaKalghatgi** [comment]
@lucianomlima Did you check?

**Comment by lucianomlima** [comment]
@SupriyaKalghatgi Yeah and I can't understand.
With Genymotion I created 5 simulators (Android versions 5.1, 6.0, 7.0, 7.1 and 8.0) and al works fine.
But in my device (Xiaomi Mi 6 with Android 8.0) and in a Moto C Plus with Android 7.0 cuts label after focus animation.

**Comment by SupriyaKalghatgi** [comment]
@lucianomlima How can i help you here?

**Comment by lucianomlima** [comment]
I know it's possible, but I don't know if it would be interesting for NativeBase.
But could have a way to pass via props the top value used in the focus animation.

So, in cases where the label is cut after animation, it could be adjusted to not cut.

**Comment by SupriyaKalghatgi** [comment]
thats something not valid logic

**Comment by lucianomlima** [comment]
Why is it not valid?
Being able to pass the top prop can help with this problem, while giving a little more flexibility.

**Comment by SupriyaKalghatgi** [comment]
That will make the use case a little complicated

**Comment by SupriyaKalghatgi** [comment]
what you suggested is kind of hack

**Comment by lucianomlima** [comment]
Ok, I understand.
Adding 2 in paddingTop for Android on lines 170 and 215 on Item component, resolves the bug here in my device.
I can send a PR with this little change. It's acceptable?

**Comment by SupriyaKalghatgi** [comment]
We can accept PR only if that doesn't cause issue on existing device types

**Comment by SupriyaKalghatgi** [comment]
sure

**Comment by lucianomlima** [comment]
PR #2149

**Comment by ZeroCool00** [comment]
i have moto g4 plus with 7.1(nougat),  and its cutting same as your image.

**Comment by SupriyaKalghatgi** [comment]
Fixed with [2.8.0](https://github.com/GeekyAnts/NativeBase/releases/tag/v2.8.0)
